Content.Server/Chemistry/Components/FoamSolutionAreaEffectComponent.cs

```diff
@@ -21,15 +22,15 @@ public class FoamSolutionAreaEffectComponent : SolutionAreaEffectComponent
         protected override void UpdateVisuals()
         {
             if (Owner.TryGetComponent(out AppearanceComponent? appearance) &&
-                SolutionContainerComponent != null)
+                EntitySystem.Get<SolutionContainerSystem>().TryGetSolution(Owner, "foam", out var solution))
```

Comment:
As elsewhere, const string. I won't spot all of these but it should be done IMO.